File: allure-java-commons/src/test/java/io/qameta/allure/examples/AllureLinksTest.java
package io.qameta.allure.examples;

import io.qameta.allure.*;
import org.junit.jupiter.api.Test;

import static io.qameta.allure.Allure.*;

public class AllureLinksTest {

    @Link("https://docs.qameta.io") // empty URL
    @Test
    public void allureOneParamLinkTest() {
        link("https://docs.qameta.io"); // empty name
    }

    @Links({@Link(name = "Allure Docs (@Link)", url = "https://docs.qameta.io"), @Link(name = "GitHub (@Link)", url = "https://github.com/allure-examples/")})
    @Test
    public void allureAnnotationLinkTest() {

    }

    @Test
    public void allureLambdaLinkTest() {
        link("Allure Docs (link)", "https://docs.qameta.io");

        // link types: tms issue custom
        link("GitHub Issue (link)", "issue", "https://github.com/allure-examples/allure-examples/issues/12");
        link("GitHub Tms (link)", "tms", "https://github.com/allure-examples/allure-examples/blob/master/allure-java-commons/README.md");
    }

    @Issues(@Issue("https://github.com/allure-examples/allure-examples/issues/12"))
    @Test
    public void allureAnnotationIssueTest() {

    }

    @Test
    public void allureLambdaIssueTest() {
        issue("GitHub Issue (issue)", "https://github.com/allure-examples/allure-examples/issues/12");
    }

    @TmsLinks(@TmsLink("https://github.com/allure-examples/allure-examples/blob/master/allure-java-commons/README.md"))
    @Test
    public void allureAnnotationTMSTest() {

    }

    @Test
    public void allureLambdaTMSTest() {
        tms("GitHub Tms (tms)", "https://github.com/allure-examples/allure-examples/blob/master/allure-java-commons/README.md");
    }

}

File: allure-python/allure-python-pytest/test/allure_links_test.py
import allure
# TODO replace test case link to TestOps


@allure.link("https://github.com/allure-examples/", name="Allure Examples")
@allure.issue("https://github.com/allure-examples/allure-examples/issues/1", name="ISSUE-1")
@allure.testcase("https://github.com/allure-examples/allure-examples/issues/2", name="TESTCASE-2")
def test_allure_links():
    pass


def test_allure_dynamic_links():
    allure.dynamic.link("https://github.com/allure-examples/allure-examples/pull/4",
                        name="Allure Examples pull request 4")
    allure.dynamic.issue("https://github.com/allure-examples/allure-examples/issues/1",
                         name="ISSUE-1")
    allure.dynamic.testcase("https://github.com/allure-examples/allure-examples/issues/2",
                            name="TESTCASE-2")


def pr(pr_id):
    return allure.link(pr_id, name=f"PR {pr_id}", link_type="pr")


@pr("123")
def test_link_templates():
    pass
